glide-core/telemetry/src/metrics_exporter_file.rs

```diff
@@ -0,0 +1,226 @@
+use async_trait::async_trait;
+use chrono::{DateTime, Utc};
+use opentelemetry_sdk::metrics::data::ResourceMetrics;
+use opentelemetry_sdk::metrics::data::{Gauge, Histogram, Sum};
+use opentelemetry_sdk::metrics::exporter::PushMetricExporter;
+use opentelemetry_sdk::metrics::MetricError;
+use opentelemetry_sdk::metrics::MetricResult;
+use opentelemetry_sdk::metrics::Temporality;
+use serde_json::{Map, Value};
+use std::any::Any;
+use std::fs::OpenOptions;
+use std::io::Write;
+use std::path::PathBuf;
+use std::sync::atomic;
+
+/// An OpenTelemetry exporter that writes Metrics to a file on export.
+pub struct FileMetricExporter {
+    is_shutdown: atomic::AtomicBool,
+    path: PathBuf,
+}
+
+impl FileMetricExporter {
+    pub fn new(path: PathBuf) -> std::result::Result<Self, MetricError> {
+        Ok(FileMetricExporter {
+            is_shutdown: atomic::AtomicBool::new(false),
+            path,
+        })
+    }
+}
+
+#[async_trait]
+impl PushMetricExporter for FileMetricExporter {
+    fn temporality(&self) -> Temporality {
+        Temporality::Cumulative
+    }
+
+    async fn export(&self, metrics: &mut ResourceMetrics) -> MetricResult<()> {
+        println!("Start to export:: metrics: {:?}", metrics);
+        if self.is_shutdown.load(atomic::Ordering::SeqCst) {
+            return Ok(());
+        }
+
+        let Ok(mut data_file) = OpenOptions::new()
+            .create(true)
+            .append(true)
+            .open(&self.path)
+        else {
+            return Err(MetricError::Other(format!(
+                "Unable to open exporter file: {} for append.",
+                self.path.display()
+            )));
+        };
+
+        let metrics_json = to_json(metrics).map_err(|e| {
+            MetricError::Other(format!("Failed to serialize metrics to JSON: {}", e))
```

Comment:
nit: on the other error mapping as well:
```suggestion
            MetricError::Other(format!("Failed to serialize metrics to JSON: {e}"))
```